doc/arch/014-reporting-etl-pipeline.md

```diff
@@ -0,0 +1,90 @@
+requirements. This reporting schema will transform the data in our application tables into a more reports-friendly
+format, exposing convenient building blocks to build reports with straightforward SQL queries.
+
+Once the reporting tables are populated, we will use Metabase to create and maintain the SQL queries that run against
+the reporting tables to generate reports.
+
+```
+ --------------------             ------------------------------------             ------------------
+| Application tables |    -->    | Reporting tables as database views |    -->    | Metabase reports |
+ --------------------             ------------------------------------             ------------------
+eg. blood_pressures              eg. blood_pressures_over_time                    eg. monthly control rate query
+                                 (latest BP for each patient as of every month)   (as of every month, how many BPs were
+                                                                                   controlled and recent)
+```
+
+The reporting tables will be implemented as database views within the Simple database.
+
+We will use Metabase for the reports.
+* Easy-to-use interface
+* Easy to grant access to stakeholders, and controlled access to ourselves if necessary
+* Dynamic queries (eg. pick your own date range)
+
+## Alternatives and constraints
+
+### Reporting tables in a separate database
+
+We could continue to generate the reporting tables by hand, but place them in a separate database. This would greatly
+alleviate performance problems. However, database performance is a secondary concern. Furthermore, the proposed approach
+will still set us up nicely to extract the reporting tables to a separate database in the future if necessary.
+
+### Rollup summary tables
+
+Rollup summary tables were proposed in [an earlier ADR](011-rollup-summary-tables-for-dashboards.md). The crux of the
+approach is to incrementally write data to these tables as application transactions occur. Eg. update a few rows in the
+"latest BPs per month" table when a new BP is recorded. The main drawback of this approach is that it's tough to iterate
+on the schemas, since every change would require a backfill of data from the beginning of time.
+
+### Third-party data warehouse like Looker or Tableau
+
+There's a high up-front cost and complexity to using a third-party tool. Additionally, there may be hidden constraints
+in such tools that we don't want to deal with while we're in an iterative stage of this effort.
+
+## Status
+
+Proposed
+
+## Consequences
+
+* We will have more nightly cron tasks to refresh the reporting views. We need to monitor the performance of these
+  refreshes closely to identify when database views are no longer a viable option for us.
+* We can start thinking of migrating existing dashboard reports to consume data from the reporting tables.
+* We will need to maintain and evolve the reporting schema as reporting requirements change.
```

Comment:
[nit] This is kind of understood, no? 😛

Reply:
Haha I meant it's and additional moving piece to maintain 😄